### PR TITLE
Switch from Angular MDC to Angular Material Fully

### PR DESCRIPTION
**Pull request by floogulinc**

### CODE REVIEW

**Comment by helloworld12321** [comment]
Oh my gosh, thanks for doing this!

File: src/app/top-menu-bar/top-menu-bar.component.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  z-index: 1;
+}
+
+.app-title {
+  padding-left: 20px;
+}
+
+.main-toolbar {
+  z-index: 5 !important;
+  will-change: transform;
```
Comment (helloworld12321) [inline src/app/top-menu-bar/top-menu-bar.component.scss:13]: I'd never heard of this one. Nifty!

File: src/app/large-display/large-display.component.html [hunk 2]
```diff
@@ -8,25 +8,17 @@
...
+    <button mat-raised-button color="accent">Quit Game</button>
   </div>
   <div class="center">
-    <button mdc-fab
+    <button mat-mini-fab color="accent"
       [disabled]="running === undefined"
       (click)="toggleTimerRunning()"
     >
-      <div *ngIf="running; then pauseIcon else playIcon"></div>
+      <mat-icon>{{running ? 'pause' : 'play_arrow' }}</mat-icon>
```
Comment (helloworld12321) [inline src/app/large-display/large-display.component.html:18]: Oh, I suppose that is simpler! :rofl:

File: src/app/top-menu-bar/top-menu-bar.component.scss (hunk 1 above)
Comment (floogulinc) [reply src/app/top-menu-bar/top-menu-bar.component.scss]: Turns out I was using that completely wrong. Don't know where I got the idea from.

Comment (helloworld12321) [reply src/app/top-menu-bar/top-menu-bar.component.scss]: Ah, roger that, then! ☺️

File: src/app/app.module.ts [hunk 3]
```diff
@@ -30,35 +26,22 @@ import { FlowerTestComponent } from './flower-test/flower-test.component';
...
-import {MdcSliderModule} from '@angular-mdc/web/slider';
-import {MdcIconButtonModule} from '@angular-mdc/web/icon-button';
-import {MdcSnackbarModule} from '@angular-mdc/web/snackbar';
 import { JsonDataTestComponent } from './json-data-test/json-data-test.component';
 
+import { MatProgressBarModule } from '@angular/material/progress-bar';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import {MatToolbarModule} from '@angular/material/toolbar';
+import {MatButtonModule} from '@angular/material/button';
+import {MatIconModule} from '@angular/material/icon';
```
Comment (kklamberty) [inline src/app/app.module.ts:35]: The spacing here should probably be consistent with other imports, which mostly have spaces between words and brackets.

File: src/app/flower-test/flower-test.component.spec.ts [hunk 4]
```diff
@@ -5,9 +5,11 @@ import { FlowerLayoutComponent } from '../flower-layout/flower-layout.component'
 import { FlowerLayoutItemComponent } from '../flower-layout-item/flower-layout-item.component';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 import { FormsModule } from '@angular/forms';
-import { MdcIconModule, MdcTopAppBarModule } from '@angular-mdc/web';
 import { TopMenuBarComponent } from '../top-menu-bar/top-menu-bar.component';
 import { FullscreenButtonComponent } from '../fullscreen-button/fullscreen-button.component';
+import {MatToolbarModule} from '@angular/material/toolbar';
+import {MatButtonModule} from '@angular/material/button';
+import {MatIconModule} from '@angular/material/icon';
```
Comment (kklamberty) [inline src/app/flower-test/flower-test.component.spec.ts:12]: Same here with the spacing

File: src/app/home/home.component.spec.ts [hunk 5]
```diff
@@ -2,7 +2,9 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { HomeComponent } from './home.component';
 import { TopMenuBarComponent } from '../top-menu-bar/top-menu-bar.component';
 import { FullscreenButtonComponent } from '../fullscreen-button/fullscreen-button.component';
-import { MdcTopAppBarModule, MdcIconModule } from '@angular-mdc/web';
+import {MatToolbarModule} from '@angular/material/toolbar';
+import {MatButtonModule} from '@angular/material/button';
+import {MatIconModule} from '@angular/material/icon';
```
Comment (kklamberty) [inline src/app/home/home.component.spec.ts:7]: Spacing

File: src/app/large-display/large-display.component.spec.ts [hunk 6]
```diff
@@ -14,6 +13,8 @@ import { TimerService } from '../timer.service';
 import { TopMenuBarComponent } from '../top-menu-bar/top-menu-bar.component';
 import { FullscreenButtonComponent } from '../fullscreen-button/fullscreen-button.component';
 import { MatProgressBarModule } from '@angular/material/progress-bar';
+import {MatButtonModule} from '@angular/material/button';
+import {MatIconModule} from '@angular/material/icon';
```
Comment (kklamberty) [inline src/app/large-display/large-display.component.spec.ts:17]: Spacing

File: src/app/timer-test/timer-test.component.spec.ts [hunk 7]
```diff
@@ -4,11 +4,13 @@ import { TimerTestComponent } from './timer-test.component';
...
 import { TimerProgressSpinnerComponent } from '../timer-progress-spinner/timer-progress-spinner.component';
 import { FormsModule } from '@angular/forms';
-import { MdcSliderModule, MdcIconModule, MdcTopAppBarModule } from '@angular-mdc/web';
 import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
 import { TopMenuBarComponent } from '../top-menu-bar/top-menu-bar.component';
 import { FullscreenButtonComponent } from '../fullscreen-button/fullscreen-button.component';
 import { MatProgressBarModule } from '@angular/material/progress-bar';
+import {MatToolbarModule} from '@angular/material/toolbar';
+import {MatButtonModule} from '@angular/material/button';
+import {MatIconModule} from '@angular/material/icon';
```
Comment (kklamberty) [inline src/app/timer-test/timer-test.component.spec.ts:13]: Spacing

File: src/app/top-menu-bar/top-menu-bar.component.spec.ts [hunk 8]
```diff
@@ -1,7 +1,9 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { TopMenuBarComponent } from './top-menu-bar.component';
-import { MdcTopAppBarModule, MdcIconModule } from '@angular-mdc/web';
 import { FullscreenButtonComponent } from '../fullscreen-button/fullscreen-button.component';
+import {MatToolbarModule} from '@angular/material/toolbar';
+import {MatButtonModule} from '@angular/material/button';
+import {MatIconModule} from '@angular/material/icon';
```
Comment (kklamberty) [inline src/app/top-menu-bar/top-menu-bar.component.spec.ts:6]: Spacing

File: src/app/app.module.ts (hunk 3 above)
Comment (kklamberty) [inline src/app/app.module.ts:35]: The spacing here should probably be consistent with other imports, which mostly have spaces between words and brackets.

File: src/app/flower-test/flower-test.component.spec.ts (hunk 4 above)
Comment (kklamberty) [inline src/app/flower-test/flower-test.component.spec.ts:12]: Spacing

File: src/app/home/home.component.spec.ts (hunk 5 above)
Comment (kklamberty) [inline src/app/home/home.component.spec.ts:7]: Spacing

File: src/app/large-display/large-display.component.spec.ts (hunk 6 above)
Comment (kklamberty) [inline src/app/large-display/large-display.component.spec.ts:17]: Spacing

File: src/app/play-round/play-round.component.html [hunk 9]
```diff
@@ -1,10 +1,10 @@
-<app-top-menu-bar [contents]="contents"></app-top-menu-bar>
```
Comment (kklamberty) [inline src/app/play-round/play-round.component.html:1]: There seem to be a couple of places where you are removing attributes similar to this one. Is that going to make testing more difficult?

File: src/app/timer-test/timer-test.component.html [hunk 10]
```diff
@@ -1,4 +1,4 @@
-<app-top-menu-bar [contents]="contents"></app-top-menu-bar>
+<app-top-menu-bar></app-top-menu-bar>
```
Comment (kklamberty) [inline src/app/timer-test/timer-test.component.html:1]: Same question as before about testing

File: src/app/play-round/play-round.component.html (hunk 9 above)
Comment (helloworld12321) [reply src/app/play-round/play-round.component.html:1]: Oh, no, that attribute was just a quirk of the old implementation :stuck_out_tongue: We were using it to make the toolbar float, but now Paul's doing that a different (better) way.

**Pull request merged**